test/e2e/storage/utils/utils.go

```diff
@@ -656,33 +656,55 @@ func genBinDataFromSeed(len int, seed int64) []byte {
 }
 
 // CheckReadFromPath validate that file can be properly read.
-func CheckReadFromPath(f *framework.Framework, pod *v1.Pod, volMode v1.PersistentVolumeMode, path string, len int, seed int64) {
+//
+// Note: disableCache only affects BlockMode PVCs and does not work with
+// (default) BusyBox Pods.
+func CheckReadFromPath(f *framework.Framework, pod *v1.Pod, volMode v1.PersistentVolumeMode, disableCache bool, path string, len int, seed int64) {
 	var pathForVolMode string
+	nocache := ""
 	if volMode == v1.PersistentVolumeBlock {
 		pathForVolMode = path
+		if disableCache {
+			// TODO: "len" (dd: bs+count) need to be adjusted to
+			// sector(s). Make sure CheckWriteToPath() wrote enough
+			// bytes!
+			len = 512
```

Comment:
Is this length change specific to RBD or will it be also required for other backend?

Reply:
This is the right size for most block devices, possibly not all. It can be detected with `blockdev --getss <dev>` though. Making sure to detect it, and write the right size in the `dd` command becomes really ugly, unfortunately.

Reply:
Should we consider changing `byteLen` in `TestKubeletRestartsAndRestoresMount` to 512?
I think that we don't need to keep comment as TODO, but we need a comment on block size consideration for block volume test case,
so that developer for any backend devices which have 512+ block size can notice it and change the size bigger.

Reply:
Modified the test to just write 512 bytes and explained the sector size in comments. I like this much better :+1: